src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Cache/CacheKeyService.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Cache;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Cache;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Cache
+{
+    public class CacheKeyService : ICacheKeyService
+    {
+        private readonly IPersistenceHelper<ContentDbContext> _contentPersistenceHelper;
+
+        public CacheKeyService(IPersistenceHelper<ContentDbContext> contentPersistenceHelper)
+        {
+            _contentPersistenceHelper = contentPersistenceHelper;
+        }
+        
+        public async Task<Either<ActionResult, DataBlockTableResultCacheKey>> CreateCacheKeyForDataBlock(Guid releaseId, Guid dataBlockId)
+        {
+            return await _contentPersistenceHelper
```

Comment:
Just wondering if you considered caching them by just data block id and not caring about the publication id or the release id to partition the cache files by publication / release? If so then I don't think this cache key service is required.

I think data blocks are cloned when creating amendments so we should be guaranteed to be caching / invalidating the right data with just a data block id I think.

I suppose partitioning them into folders by release does make identifying them easier and also makes it easier to clean them up if a release is deleted.

Reply:
I figured that it might be best to stick with the general convention of the publication / release paths for now until we decided one way or the other whether to move them wholesale to just Guids.  It does make deleting a whole Amendment's-worth of cached content a little easier during cancellation I guess though (see latest commit!).

Reply:
Cool, thought that might be the case, thanks. See my comment about tearing down data which it probably also helps with.